sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpMessageConverterTests.cs

```diff
@@ -812,9 +812,7 @@ public void CreateEventFromMessageValidatesTheSource()
         public void CreateEventFromMessagePopulatesTheBody()
         {
             var body = new byte[] { 0x11, 0x22, 0x33 };
-
-            using var bodyStream = new MemoryStream(body, false);
-            using var message = AmqpMessage.Create(bodyStream, true);
+            using var message = AmqpMessage.Create(new Data { Value = body });
```

Comment:
Though the stream-based `Create` method serializes to a `Data` section for transport, it does not create the internal data list, but rather a specialized [`AmqpBodyStreamMessage`](https://github.com/Azure/azure-amqp/blob/f9d7c10a6c77dfcc0189829c1f6a51abd8c45592/src/AmqpMessage.cs#L894) from which the `BodyData` cannot be accessed directly, despite having the section flag set.

Now that we're no longer reading the body as a stream when converting from `AmqpMessage` => `EventData`, the approach used in these tests no longer worked.   The new approach is consistent with how we do conversion from `EventData` => `AmqpMessage` as well as the form expected when reading events published using the legacy client library.